### DISCUSSION TITLE
Generation of a proper mechanism file

### DISCUSSION THREAD

**Discussion by erangit** [mutationpp/Mutationpp#174] [Q&A]
I'm learning to use Mutationpp through SU2-NEMO coupling
I need to create a mixture of humid air. 
No problem to create the mixture file ([N N2 O O2 H2O H H2 Ho]
But I have no idea how to create the mechanism file (not the format but which reactions,
their constants, and so forth). 
In Cantera, I used to use published mechanisms (in chemkin or cti format)
by places like UCSD, UMINN , Aramco to create the mechanism file. Is there a parallel here?
Sorry if the issue is in the wrong place
Eran

**Comment by jbscoggi** [comment]
Hey @erangit, thank you for the question.  I think the simple answer is that there is no need to mix the formatting of the reactions in the mechanism with the reactions/data that you use in the mechanism.  In other words, if you use published mechanisms that you are familiar with before, there is no reason not to use the same mechanism in M++.  The only issue is that you may need to reformat the data into the M++ mechanism format, but this is only a formatting issue rather than a question of the fundamental physics you want to model.  This should not change, regardless of the library you use to evaluate the mechanism.

With that said, you might be reluctant to try and reformat the mechanism file from the chemkin or cti format.  At one point I had a conversion script from chemkin to M++, but unfortunately I can't seem to find it.  I definitely think this would be a useful tool to add in the scripts folder for M++, but for the moment I don't have the bandwidth to work on this.  Perhaps someone who sees this thread could do it, or you might be willing to write it?  If so, we would definitely welcome the contribution!

I hope that helps, let me know if it's still not clear.
